src/fly-to-viewport.js

```diff
@@ -77,4 +58,45 @@ export default function flyToViewport(startProps, endProps, t) {
   viewport.zoom = newZoom;
   return viewport;
 }
-/* eslint-enable max-statements */
+
+// returns transition duration in milliseconds
+export function getFlyToDuration(startProps, endProps, opts = {}) {
+
+  const {S} = getFlyToTransitionParams(startProps, endProps);
+  const {speed} = opts;
+
+  return 1000 * S / (speed || DEFAULT_PROPS.speed);
```

Comment:
What is the unit of `S`? Shall we just export a `getFlyToDistance` and let react-map-gl determine what options to expose?

Reply:
`S` is an intermediate value produced based on zoom and position (lng/lat) change, there is no specific unit for this.

Reply:
You either need to explain `S` or explain `speed`, which does not have more clarity...

Reply:
Updated to return `S`,  changed util method name to :`getFlyToLength`.